src/Output/MultipleFileOutput.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+namespace Spatie\TypeScriptTransformer\Output;
+
+use Spatie\TypeScriptTransformer\TypeScriptTransformerConfig;
+
+class MultipleFileOutput extends Output
+{
+    public function __construct(protected TypeScriptTransformerConfig $config)
+    {
+        $path = $config->getOutputDestination();
+        $this->path = is_dir($path) ? rtrim($path, '/') : pathinfo($path, PATHINFO_DIRNAME);
+        $this->output = [];
+
+    }
+
+    public function append(string $typeScript, string $classPath = null): void
+    {
+        $pathName = ltrim(str_replace('\\', '/', $classPath), '/');
+
+        $this->output[$pathName] = $typeScript;
+    }
+
+    public function writeOut($fileType): array
```

Comment:
We don't need to return an array because the returned value is never used. Also let's call this method `writeFiles` that's a bit more clear on what it is actually doing